### ISSUE TITLE
Next 12 Build Performance on Windows

### ISSUE THREAD

**Issue by lukaszfiszer** [vercel/next.js#31021]
### What version of Next.js are you using?

12.0.2 and 12.0.3-canary.9

### What version of Node.js are you using?

v14.15.4

### What browser are you using?

Any

### What operating system are you using?

Windows

### How are you deploying your application?

N/A

### Describe the Bug

Despite the announcement (https://nextjs.org/blog/next-12#faster-builds-and-fast-refresh-with-rust-compiler) the build time of Next12 app on Windows was not improved:

- empty application created with `create-next-app@latest` - the build is the same comparing to `create-next-app@11` (approx. 7.8 seconds)
- small-size application (5 pages, 2 API endpoints) - the build is ~50% slower after updating with `npm i next@latest` by 50% (22.4s vs 32.0)

Others have reported similar or even worse results: https://www.tpjnorton.com/blog/posts/next-js-12-performance-test-builds-appear-t o-be-slower-not-faster

Upgrading to `next@canary` makes no different.

### Expected Behavior

Faster build with next12 comparing to next11.

### To Reproduce

In Windows Powershell run:

```
npx create-next-app@latest
cd app-name
Measure-Command  {npm run build}
Measure-Command  {npm run build}
```

Compare results with:

```
npx create-next-app@11
cd app-name
Measure-Command  {npm run build} 
Measure-Command  {npm run build} 
```

Both build commands (with cold and hot cache) execution times are +/- the same.

**Comment by jamsinclair** [comment]
Seeing slower build times (30% increase) for a medium application after upgrading to Next 12.0.3 on Mac OS too.

#### Cold build times (with `next build` and  type checking disabled)
- Next 11.1.0 **~30 seconds**
- Next 12.0.3 **~39 seconds**

#### App
- 6 API Routes
- 2 base ISR Pages (hundreds of pages are later dynamically built from here)
- Uses TypeScript
- Uses `next-transpile-modules` for one 200KB dependency  
- ~Uses `next-with-less`~ I've removed this plugin and still observing ~10 second increase

Would a build trace help provide more visibility? Happy to share privately with a maintainer.

**Comment by amesas** [comment]
Same here, 12.x is slower than 11.1

**Comment by olaj** [comment]
Same here, I'm on Windows. Builds are much slower, on multiple computers, and also tried in WSL, same result. Don't have time atm to dig deeper but well, just wanted to say i have the issue as well.

**Comment by timneutkens** [comment]
Please provide the `.next/trace` file both before/after and we'll have a look 👍 I'm thinking it might be related to outputFileTracing being a slowdown in the particular app or particularly on windows 🤔

**Comment by olaj** [comment]
@timneutkens My project is closed source and i can be a bit sloppy with secrets and stuff, are there anything secret in the trace files? Can i e-mail them somewhere? Or can i be 100 safe sharing them? 😨 

I just ran now and the first build with Next 11 took about 36 sec, first build on Next 12 about 56 sec, second build took 10 sec on Next 11 and about 36 or something on Next 12. So its a major diffrence.

**Comment by tbgse** [comment]
We're also seeing much slower builds on 12.x, build time on vercel is up from 5 to 20 minutes, dev server is barely useable. Also happy to share files / code privately if that helps (created a support ticket on vercel too)

**Comment by timneutkens** [comment]
@olaj there's no secrets/code in the trace file, only timings 👍 

@tbgse happy to take a look at the app if that's available? Can you send the `.next/trace` as well?

**Comment by tbgse** [comment]
@timneutkens yeah happy to send through the files for builds both before & after. Since they're both pretty large, do you have a place where i can send them to?

**Comment by timneutkens** [comment]
Generally GitHub gist works for sending them, but any other way is fine too, feel free to email to Vercel support saying they're for me and I'll get them as well 👍

**Comment by tbgse** [comment]
Ok, just sent them through to Vercel support, thanks for looking into this! The app itself is private but if the logs get you curious I'd be happy to grant you access or walk you through it.

**Comment by olaj** [comment]
I've e-mailed as well now!

**Comment by lukaszfiszer** [comment]
@timneutkens I've uploaded build traces of empty application (generated using `create-next-app`) to [gist: https://gist.github.com/lukaszfiszer/9b3351645c2b7296c14c9266d4a5aa7c](https://gist.github.com/lukaszfiszer/9b3351645c2b7296c14c9266d4a5aa7c)

**Comment by timneutkens** [comment]
@lukaszfiszer both of those traces are running Next.js 12.0.4 🤔

**Comment by amesas** [comment]
> @lukaszfiszer both of those traces are running Next.js 12.0.4 🤔

12.0.3 had the same prob for sure. I have not done a real test, but I've the feeling that all 12.x releases has been slower.

**Comment by sfiquet** [comment]
@amesas The point is that he needs a version 11 trace and a version 12 trace to compare them.

**Comment by jamsinclair** [comment]
@timneutkens I have a [gist of the Next 11 vs Next 12 traces](https://gist.github.com/jamsinclair/80f0ff35c08258b2a231a4668c862b73) if it's any help. Note: I've sanitised some of the module paths.

Thanks! 🙇

**Comment by timneutkens** [comment]
@jamsinclair @olaj  for some reason output file tracing is slower than expected on your app, you can set `outputFileTracing: false` which disables tracing files during `next build`.

**Comment by jamsinclair** [comment]
Thanks for the update! Yes, can confirm, setting `outputFileTracing: false` results in a much faster build with Next 12. Down from average 38 seconds to 28 seconds.

**Comment by TheThirdRace** [comment]
> outputFileTracing: false

This also "solved" my problem. I'm on Window 10 too so you might have found the pattern.

The documentation on `outputFileTracing: false` make it sound scary to disable this, like the build could be missing files and not work... Should I be worried?

While my repo is private, I could send an invite for you to check it out. Furthermore, I'm in the midst of migrating from `11.1.2` to `12.0.7` so I could easily make 2 branches for testing purpose: `nextjs-11-1-2` and `next-12-0-7`.

Let me know if it's something that you're interested in validating.

## Results of `yarn build`

### 11.1.2
Cold build: `~25s`
Warm build: `~14s`

### 12.0.7
Cold build: `~35s` (+10s average)
Warm build: `~24s` (+10s average)

### 12.0.7 + `outputFileTracing: false`
Cold build: `~19s` (-6s average vs 11.1.2)
Warm build: `~10s` (-4s average vs 11.1.2)

**Comment by olaj** [comment]
Sorry, had missed the updated in here.

I see similar numbers, in one of my projects (with material-ui) it went from 36s to 14s or something like that. In my other project that haven't had that much issues with it went from 16s to 10s.

I could also invite you to a project if you like to!

**Comment by TheThirdRace** [comment]
## Results with Canary

| outputFileTracing | cold build | warm build |
|:--|--:|--:|
| true | 31.44s | 18.40s |
| false | 17.29s | 8.37s |

\* cold build = no `.next/` folder before build
\* warm build = `.next/` folder exists before build

Needless to say, Windows still has problems with the `outputFileTracing = true` option...

**Comment by TheThirdRace** [comment]
According to a new message in NextJs `14.x`:

![image](https://github.com/vercel/next.js/assets/22915526/b9c5da2f-a430-451d-b37d-1a36b087aee7)

🚨 Please DO NOT remove this option 🚨

This is still 100% necessary to disable it for Windows, even if everything is located within the WSL2. Removing that option makes my build take 2 to 3 times as long and my SSD is no slouch...

Unfortunately, this bug was raised more than 2 years ago and hasn't been solved... So if it's not getting fixed definitively for Windows, don't remove the option.

**Comment by TheThirdRace** [comment]
Issue is still relevant, been waiting almost 4 years for it...
